src/js/components/FocusedContainer.js

```diff
@@ -18,44 +13,56 @@ export const FocusedContainer = ({
   const [bodyOverflowStyle, setBodyOverflowStyle] = useState('');
   const ref = useRef(null);
 
+  const roots = useContext(RootsContext);
+  const [nextRoots, setNextRoots] = useState(roots);
   useEffect(() => {
-    const removeTrap = () => {
-      const child = ref.current;
-      getBodyChildElements()
-        .filter(isNotAncestorOf(child))
-        .forEach(makeNodeFocusable);
-      if (restrictScroll) {
-        document.body.style.overflow = bodyOverflowStyle;
-      }
-    };
+    // make sure value of null is not added to array
+    if (ref.current) setNextRoots([...roots, ref.current]);
+  }, [roots]);
 
-    const handleTrapFocus = () => {
-      const child = ref.current;
-      getBodyChildElements()
-        .filter(isNotAncestorOf(child))
-        .forEach(makeNodeUnfocusable);
+  useEffect(() => {
+    if (
+      bodyOverflowStyle !== 'hidden' &&
+      !hidden &&
+      restrictScroll &&
+      trapFocus
+    ) {
+      setBodyOverflowStyle(document.body.style.overflow);
+      document.body.style.overflow = 'hidden';
+    }
 
-      if (restrictScroll && bodyOverflowStyle !== 'hidden') {
-        setBodyOverflowStyle(document.body.style.overflow);
-        document.body.style.overflow = 'hidden';
+    return () => {
+      if (
+        bodyOverflowStyle !== 'hidden' &&
+        !hidden &&
+        restrictScroll &&
+        trapFocus
+      ) {
+        document.body.style.overflow = bodyOverflowStyle;
       }
     };
+  }, [bodyOverflowStyle, hidden, trapFocus, restrictScroll]);
 
+  useEffect(() => {
     const timer = setTimeout(() => {
-      if (!hidden && trapFocus) {
-        handleTrapFocus();
+      if (!hidden && trapFocus && roots && roots[0] !== null) {
+        roots.forEach(makeNodeUnfocusable);
       }
     }, 0);
 
     return () => {
-      removeTrap();
+      // remove trap and restore ability to focus on the last root only
```

Comment:
Question/Wondering: This is a style related comment, but since useEffects aren't named you don't get the benefit of having a clue of what's happening inside like you do with a well named function. 

The comment here is helpful, but I have been wondering about moving them just ahead of the useEffect so the reader knows why the effect is being run sooner.

In this particular instance, its positioning isn't really an issue, but when effects have more going on it can be quite helpful.

Just a wondering... / thought to share.

Reply:
that's a good point -- in this case this comment is a little more specific to the clean-up function but I agree that a comment at the start of this effect could be helpful to say "toggle aria-hidden value for roots", and a similar pattern adopted in future